pkg/providers/aws/provider_postgres.go

```diff
@@ -0,0 +1,487 @@
+		listOutput, err := cacheSvc.DescribeDBInstances(&rds.DescribeDBInstancesInput{})
+		if err != nil {
+			return false, nil
+		}
+		pi = listOutput.DBInstances
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return pi, nil
+}
+
+func (p *AWSPostgresProvider) getPostgresConfig(ctx context.Context, r *v1alpha1.Postgres) (*rds.CreateDBInstanceInput, *rds.DeleteDBInstanceInput, *StrategyConfig, error) {
+	stratCfg, err := p.ConfigManager.ReadStorageStrategy(ctx, providers.PostgresResourceType, r.Spec.Tier)
+	if err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to read aws strategy config")
+	}
+	if stratCfg.Region == "" {
+		stratCfg.Region = DefaultRegion
+	}
+
+	postgresCreateConfig := &rds.CreateDBInstanceInput{}
+	if err := json.Unmarshal(stratCfg.RawStrategy, postgresCreateConfig); err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to unmarshal aws postgres cluster configuration")
+	}
+
+	postgresDeleteConfig := &rds.DeleteDBInstanceInput{}
+	if err := json.Unmarshal(stratCfg.DeleteStrategy, postgresDeleteConfig); err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to unmarshal aws postgres cluster configuration")
+	}
+	return postgresCreateConfig, postgresDeleteConfig, stratCfg, nil
+}
+
+// verifies if there is a change between a found instance and the configuration from the instance strat
+func verifyPostgresChange(postgresConfig *rds.CreateDBInstanceInput, foundConfig *rds.DBInstance) *rds.ModifyDBInstanceInput {
+	updateFound := false
+
+	mi := &rds.ModifyDBInstanceInput{}
+	mi.DBInstanceIdentifier = postgresConfig.DBInstanceIdentifier
+
+	if *postgresConfig.DeletionProtection != *foundConfig.DeletionProtection {
+		mi.DeletionProtection = postgresConfig.DeletionProtection
+		updateFound = true
+	}
+	if *postgresConfig.Port != *foundConfig.Endpoint.Port {
+		mi.DBPortNumber = postgresConfig.Port
+		updateFound = true
+	}
+	if *postgresConfig.BackupRetentionPeriod != *foundConfig.BackupRetentionPeriod {
+		mi.BackupRetentionPeriod = postgresConfig.BackupRetentionPeriod
+		updateFound = true
+	}
+	if *postgresConfig.DBInstanceClass != *foundConfig.DBInstanceClass {
+		mi.DBInstanceClass = postgresConfig.DBInstanceClass
+		updateFound = true
+	}
+	if *postgresConfig.PubliclyAccessible != *foundConfig.PubliclyAccessible {
+		mi.PubliclyAccessible = postgresConfig.PubliclyAccessible
+		updateFound = true
+	}
+	if *postgresConfig.AllocatedStorage != *foundConfig.AllocatedStorage {
+		mi.AllocatedStorage = postgresConfig.AllocatedStorage
+		updateFound = true
+	}
+	if *postgresConfig.EngineVersion != *foundConfig.EngineVersion {
+		mi.EngineVersion = postgresConfig.EngineVersion
+		updateFound = true
+	}
+	if !updateFound {
+		return nil
+	}
+	return mi
+}
+
+// verify postgres create config
+func (p *AWSPostgresProvider) verifyPostgresCreateConfig(ctx context.Context, pg *v1alpha1.Postgres, postgresCreateConfig *rds.CreateDBInstanceInput) error {
```

Comment:
The func name here suggests to me there'd be an error if `pg` or `postgresCreateConfig` was invalid. `buildPostgresCreateStrategy` might be a more descriptive name. wdyt?